Assignment4/Mariogarber-21c025/Task_07.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+"""Copia de Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1w7Dstb5eU5F7wG9x48k0ckL5fDbXYz-w
+
+**Task 07: Querying RDF(s)**
+"""
+
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage + "/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+
+schema = Namespace('http://www.w3.org/2000/01/rdf-schema#')
+
+q1 = prepareQuery('''
+  SELECT ?SubClass WHERE { 
+    ?SubClass schema:subClassOf <http://somewhere#Person>.
+  }
+  ''',
+                  initNs={"schema": schema}
+                  )
+# Visualize the results
+for r in g.query(q1):
+    print(r.SubClass)
+
+ns = Namespace("http://somewhere#")
+
+Subclass = g.triples((None, schema.subClassOf, ns.Person))
+for s, p, o in Subclass:
+    print(s)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+
+syntax = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#')
+
+q2 = prepareQuery('''
+   SELECT ?Subject WHERE { 
+    {?Subject schema:type ns:Person} UNION 
+    {?Subject schema:type ?tipo.
+    ?tipo syntax:subClassOf ns:Person}
```

Comment:
use subClassOf* to get all the possible subclasses of Person (including subclasses of subclasses)